### ISSUE TITLE
Ads notification shows up during DnD time

### ISSUE THREAD

**Issue by srirambv** [brave/browser-android-tabs#1902]
## Description 
Ads notification shows up during DnD time

## Steps to reproduce 

Pre-requisite: Ads per day cap should not be reached. Best case don't use browser during the day and only start using after 9pm. 

   1. Use the browser for a while after 9pm 
   2. Visit sites that will trigger ads notification
   3. Put the app in background and lock screen, ad notification will show up 

## Actual result 
Ad notification showed up when the device got locked around 9.53 which is well past the DnD time

## Expected result
No ad notification to be shown for 2 reasons 
1. Its past DnD time 
2. App was active but before the device got locked it was backgrounded

## Issue reproduces how often 
Only happened once. Will check if it consistently reproduces otherwise good luck trying to reproduce

## Issue happens on 
   - Current Play Store version? Yes
   - Beta build? N/A

## Device details
   - Install type (ARM, x86): ARM
   - Device (Phone, Tablet, Phablet): Pixel 3XL
   - Android version: 9.1
   
## Brave version
1.1.2

### Website problems only
- Does the issue resolve itself when disabling Brave Shields? 
- Is the issue reproducible on the latest version of Chrome? 

### Additional information

cc: @tmancey @jsecretan @mandar-brave

**Comment by mandar-brave** [comment]
@srirambv this is software; if the browser asked for a notification while it was running in foreground and you clicked the lock button, the call to render notification goes to the system which then triggers notification at an optimal time. 
If there was a significant time delay (1+ min), then the issue is worth looking at. if it is instantaneous, then it is not clear if there is a real fix. App cannot predict if user decides to lock screen!

@tmancey what is your take.

**Comment by srirambv** [comment]
>If there was a significant time delay (1+ min), then the issue is worth looking at. if it is instantaneous

It wasn't like the moment i locked the device the notification showed up. There was a slight delay may be not  1+min but close to a min

**Comment by mandar-brave** [comment]
We can expect a delay, the notification delegate is the system so unclear if we can have that sort of a granular control. I will let Terry comment!

**Comment by srirambv** [comment]
@mandar-brave @tmancey @BrendanEich I believe there is an issue with notifications showing up during DND. I was able to reproduce the issue again but unfortunately, it happened when I was not at the desk so couldn't capture the logs. I believe this scenario is possible if the daily cap of ads isn't met. So what I did was 
* Not use the browser during the day so that daily cap isn't met
* After 9 PM, use the browser for a bit
* Let the device lock itself 
* 2 ad notification showed up after 8-9mins when the device was allowed to auto-lock 

So definitely the issue is reproducible. _IF_ there are a few ads already viewed then logs do show up as notification not made which was verified during the release.

**Comment by tmancey** [comment]
@gdregalo Can you please confirm `IsNotificationsAvailable` function for `notifications_helper_android.cc` would not return `true` if device is set to `Do Not Disturb`, so this function should only return `true` if ad notifications should be shown

**Comment by mandar-brave** [comment]
also @srirambv is the TZ on your device in -- in sync with your habitat?

**Comment by tmancey** [comment]
@srirambv I have asked @gdregalo to check the Android side (or if he is busy maybe another dev can check), but I am not seeing this issue. Could you please send a screenshot of settings on the device related to timezones and locale (as a follow up to @mandar-brave question)

**Comment by srirambv** [comment]
> also @srirambv is the TZ on your device in -- in sync with your habitat?

Yes. No change as this is on my daily driver and not on the test device

> Could you please send a screenshot of settings on the device related to timezones and locale (as a follow up to @mandar-brave question)
![Screenshot_20190807-122439](https://user-images.githubusercontent.com/17010094/62639919-7bd8d780-b90e-11e9-8115-bbfc0e3b20e3.png)
![Screenshot_20190807-122501](https://user-images.githubusercontent.com/17010094/62639920-7bd8d780-b90e-11e9-840f-bde35d77601f.png)

**Comment by gdregalo** [comment]
It seems there is a confusion here between Android `Do Not Disturb mode` (DND)  (https://developer.android.com/guide/topics/ui/notifiers/notifications.html#dnd-mode) 
and `Ads-free time` logic implemented by browser.
DND works as expected: it silences sounds and vibration for all notifications but still let them to go through.

**Comment by gdregalo** [comment]
https://bravesoftware.slack.com/archives/CJVNBPK9P/p1565371165112100
